### ISSUE TITLE
regrid_shape and masked arrays

### ISSUE THREAD

**Issue by DamienIrving** [SciTools/cartopy#1002]
I've successfully used the `regrid_shape` keyword on numerous occasions when creating [quiver](http://scitools.org.uk/cartopy/docs/v0.11/matplotlib/geoaxes.html#cartopy.mpl.geoaxes.GeoAxes.quiver) plots. However, today I tried to plot the ocean surface current (i.e. a masked array where the land is masked) and it didn't work (it plotted very large quivers over land and vanishingly small over the ocean). 

If the `regrid_shape` keyword isn't designed to handle masked arrays, is there a way around this issue. Perhaps set all the land values to zero?

**Comment by ajdawson** [comment]
I think the code expects numpy arrays and will use them as such, which for masked arrays means it will use the fill value for missing points. You can try filling the component arrays with zeros (i.e. `my_masked_array.filled(0)`) before plotting them as a workaround.

**Comment by DamienIrving** [comment]
@ajdawson Yep, it definitely used the fill values. Using zero was pretty good but still some weirdly large values near coastlines. I found the best solution was to fill with `my_masked_array.filled(np.nan)` before plotting.

**Comment by pelson** [comment]
Thanks for raising @DamienIrving. I've added this as an enhancement request, as it would be good to handle masked arrays. Hopefully a pretty straightforward PR...
